dev/.rat-excludes

```diff
@@ -101,3 +101,4 @@ org.apache.spark.scheduler.ExternalClusterManager
 .*\.sql
 .Rbuildignore
 org.apache.spark.deploy.yarn.security.ServiceCredentialProvider
+spark-warehouse
```

Comment:
@frreiss this seems fine but better still is to understand what's leaving this lying around. You can see all the debris that various tests have been leaving around by looking at this file by somewhere just doing a `new File("mytestdir")`